docs/FORD.OpenATF.SDD.TPL.md

```diff
@@ -0,0 +1,229 @@
+Example:
+     ![State diagram](../../../modules/docs/assets/validation_state_diagram.svg)
+     or
+     *<!Component Name> has no specific states to declare.*
+<!!!!!!!!!!!END_INSTRUCTION!!!!!!!!!!!!!>
+
+For more information about class digram follow:
+
+- <http://www.uml-diagrams.org/state-machine-diagrams.html>
+
+## 3. Component Interfaces
+## 3.1 Public interfaces description
+<!Component Name> provides functionality with following interfaces:
+
+<!!!!!!!!!!!BEGIN_INSTRUCTION!!!!!!!!!!!!!>
+Here need to be added a list of external interfaces
+Example:
+    - `load_schema`
+    - `schema_validation`
+Note: Text surrounded by ` signs will be auto-added by ldoc
+<!!!!!!!!!!!END_INSTRUCTION!!!!!!!!!!!!!>
+
+## 3.2 Internal interfaces description
+The following interfaces are provided by <!Component Name> for internal usage only:
+
+<!!!!!!!!!!!BEGIN_INSTRUCTION!!!!!!!!!!!!!>
+Here need to be added a list of external interfaces
+Example:
+    - `api_loader`
+Note: Text surrounded by ` signs will be auto-added by ldoc
+<!!!!!!!!!!!END_INSTRUCTION!!!!!!!!!!!!!>
+
+
+## 3.3 Derived interfaces and dependencies
+<!Component Name> requires following modules:
+
+<!!!!!!!!!!!BEGIN_INSTRUCTION!!!!!!!!!!!!!>
+Here need to be added a list of libraries or other components
+Example:
+    -  OpenSSL library v 1.0.1g and higher to meet TLS cipher restricts
+    - `xml`
+    - `json`
+<!!!!!!!!!!!END_INSTRUCTION!!!!!!!!!!!!!>
+
+## 4. Component data and resources
+## 4.1 Element Data Structure
+<!Component Name> provides following data:
+
+<!!!!!!!!!!!BEGIN_INSTRUCTION!!!!!!!!!!!!!>
+Here need to be added a list of component data types
+Example:
+    - API schema from the xml file provided by `api_loader.init`
+    - RPC validation table are provided by `schema_validation.CreateSchemaValidator`
+All link will be auto-added by doxygen
+<!!!!!!!!!!!END_INSTRUCTION!!!!!!!!!!!!!>
+
+## 4.2 Resource usage
+
+<!!!!!!!!!!!BEGIN_INSTRUCTION!!!!!!!!!!!!!>
+Here need to be added all resource-related information
+All file, database or network reading
+An amount of processing by component data
+Example:
+     Resumption uses QBD/JSON database with configurable limitation 10 Mb
+     Request Controller Handle a configured amount of RPCs:
+       - A XXX count of messages from application in NONE level
+              + <LINK_TO_REQURMENT_XXX>
+       - A YYY count of messages per second for each application
+              + <LINK_TO_REQURMENT_YYY>
+ (!) In case of no such restrict it need to be clarified  (!)
+<!!!!!!!!!!!END_INSTRUCTION!!!!!!!!!!!!!>
+
+## 5.1 References
+<!!!!!!!!!!!BEGIN_INSTRUCTION!!!!!!!!!!!!!>
+Here need to be added a list of all related to component functionality
+references, including 3d-party libraries, documentation, requirements 
+Example:
+  -  [OpenSSL API](https://www.openssl.org/docs/manmaster/ssl/)
+  -  [SQLite Documents](https://www.sqlite.org/docs.html)
+<!!!!!!!!!!!END_INSTRUCTION!!!!!!!!!!!!!>
+
+## 5.2 Document history
```

Comment:
@LuxoftAKutsan,
If you will check next line it wiill be a github history ;)
Note: For external users: as support engineers, QCA and so on it will not be clear **from the final html** where to find a history - that why this chapter is here.